### PR TITLE
feat(2676): Support passing in full path for getFile [1]

### PR DESCRIPTION
**Pull request by tkyi**
## Context

In order to support getting provider config dynamically in screwdriver.yaml, we need to be able to pass in the full path to getFile instead of the scmUri, etc.

## Objective

This PR changes getFile method to allow using full path to get repo owner, name, branch, etc info.

## References

https://github.com/screwdriver-cd/screwdriver/issues/2676
## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Review by jithine — APPROVED** [review]

File: index.js [hunk 1]
```diff
@@ -1112,17 +1113,29 @@ class GithubScm extends Scm {
      * @return {Promise}                      Resolves to string containing contents of file
      */
     async _getFile({ scmUri, path, token, ref, scmRepo }) {
-        const lookupConfig = {
-            scmUri,
-            token
-        };
+        let fullPath = path;
```
Comment (pritamstyz4ever) [inline index.js:1116]: can we also override the provider in jobs? If yes, can we override the properties individually?

**Review by pritamstyz4ever — APPROVED** [review]
Also, need to update the docs
https://github.com/screwdriver-cd/guide/blob/master/docs/cluster-management/aws-integration.md

File: index.js (hunk 1 above)
Comment (tkyi) [reply index.js:1116]: I'll look into it in the models/config-parser repos
I think we'll need to implement something similar to this to allow merging of individual properties under provider: 
https://docs.screwdriver.cd/user-guide/templates#merging-with-shared-steps

**Comment by tkyi** [comment]
Will need to port this change over to scm-bitbucket and scm-gitlab later too

**Pull request merged**